ddpui/api/airbyte_api.py

```diff
@@ -1,748 +1,329 @@
-def get_airbyte_destination_definitions(request):
-    """Fetch destination definitions in the user organization workspace"""
-    orguser: OrgUser = request.orguser
-    if orguser.org.airbyte_workspace_id is None:
-        raise HttpError(400, "create an airbyte workspace first")
-
-    res = airbyte_service.get_destination_definitions(orguser.org.airbyte_workspace_id)[
-        "destinationDefinitions"
-    ]
-    allowed_destinations = os.getenv("AIRBYTE_DESTINATION_TYPES")
-    if allowed_destinations:
-        res = [destdef for destdef in res if destdef["name"] in allowed_destinations.split(",")]
-    logger.debug(res)
-    return res
-
-
-@airbyte_router.get(
-    "/destination_definitions/{destinationdef_id}/specifications",
-    auth=auth.CustomAuthMiddleware(),
-)
-@has_permission(["can_view_warehouse"])
-def get_airbyte_destination_definition_specifications(request, destinationdef_id):
+    try:
+        org: Org = Org.objects.get(id=orgid)
+        airbyte_service: AirbyteService = AirbyteService(org)
+        source_definitions: List[Dict[str, Any]] = airbyte_service.list_source_definitions()
+        return JsonResponse({"source_definitions": source_definitions})
+    except Exception as e:
+        logger.error(f"get_source_definitions: {str(e)}")
+        return JsonResponse({"error": str(e)}, status=500)
+
+def get_source_definition_specifications(request: Request, orgid: str, source_definition_id: str) -> JsonResponse:
     """
-    Fetch specifications for a destination
-    definition in the user organization workspace
+    Get specifications for a specific source definition.
+    
+    Args:
+        request: The HTTP request object
+        orgid: The organization ID
+        source_definition_id: The source definition ID
+        
+    Returns:
+        JsonResponse containing source definition specifications
     """
-    orguser: OrgUser = request.orguser
-    if orguser.org.airbyte_workspace_id is None:
-        raise HttpError(400, "create an airbyte workspace first")
-
-    res = airbyte_service.get_destination_definition_specification(
-        orguser.org.airbyte_workspace_id, destinationdef_id
-    )["connectionSpecification"]
-    logger.debug(res)
-    return res
-
-
-@airbyte_router.post("/destinations/", auth=auth.CustomAuthMiddleware())
-@has_permission(["can_create_warehouse"])
-def post_airbyte_destination(request, payload: AirbyteDestinationCreate):
-    """Create an airbyte destination in the user organization workspace"""
-    orguser: OrgUser = request.orguser
-    if orguser.org.airbyte_workspace_id is None:
-        raise HttpError(400, "create an airbyte workspace first")
-
-    destination = airbyte_service.create_destination(
-        orguser.org.airbyte_workspace_id,
-        payload.name,
-        payload.destinationDefId,
-        payload.config,
-    )
-    logger.info("created destination having id " + destination["destinationId"])
-    return {"destinationId": destination["destinationId"]}
-
```

Comment:
Can you run pre-commit locally if you haven't already ? I see some format changes (i maybe wrong)